### PR TITLE
display shift status in pdf export

### LINKED ISSUE #88: improve PDF export for SectionBasedSignupMethod
As a planner, I'd like to see the positions and their required qualifications in the PDF export. If the position is already taken, there should be the name of the user next to it. See the PDF linked in #10 for reference. Probably we want the signup methods to deliver something like a dict that is rendered to the table in the PDF file.

### PR DESCRIPTION
**Pull request by jeriox**
closes #88 

TODO
- [x] remove helper code once #343 is merged
- [x] figure out a way to use `get_participation_display()` of InstantConfirmationSignupMethod on RequestConfirmSignupMethod
- [x] improve layout (padding of table, borders)
- [x] render sections with no participants

### CODE REVIEW

File: ephios/plugins/basesignup/signup/common.py [hunk 1]
```diff
@@ -102,3 +103,36 @@ def get_configuration_fields(self):
...
+                ", ".join(
+                    participant.qualifications.filter(
+                        category__in=relevant_qualification_categories
+                    ).values_list("title", flat=True)
+                ),
+            ]
+            for participant in self.shift.get_participants()
+        ]
+        try:
+            # we may not be using MinMaxParticipantsMixin so there may be no bounds
```
Comment (felixrindt) [inline ephios/plugins/basesignup/signup/common.py:123]: Well that comment is not completely right. The Mixin only provides one way to implement `get_participant_count_bounds`. SectionBasedSignup doesn't use it and still implements the method.

File: ephios/core/signup/methods.py [hunk 2]
```diff
@@ -378,6 +378,15 @@ def render_shift_state(self, request):
         """
         return ""
 
+    def get_participation_display(self):
+        """
+        Returns a displayable representation of participation (list of lists) that can be rendered into a table.
+        """
```
Comment (felixrindt) [inline ephios/core/signup/methods.py:384]: As I see it, this data will mainly be used for pdfs, what else? Could better reflect that in the method name.
Also, I'd like to see the docstring go into more detail about the format and data to include.

File: ephios/plugins/basesignup/signup/common.py (hunk 1 above)
Comment (felixrindt) [reply ephios/plugins/basesignup/signup/common.py:123]: And plugins might use the QualificationMixin but not the MinMaxMixin but their own way...

File: ephios/core/signup/methods.py (hunk 2 above)
Comment (jeriox) [reply ephios/core/signup/methods.py:384]: yeah i wasn't really sure about how the separation should be done here. I started by building the section of the PDF completely in the signup method, but this isn't perfect as we don't know to which pagesize the data will be rendered. so I switched over to this approach where the method only delivers a list that *could* be used elsewhere in the future (say some kind of dashboard or whatever). this is why I didn't name the method specifically for pdf usage

Comment (felixrindt) [reply ephios/core/signup/methods.py:384]: If you could elaborate on the "lists of lists" a little in the docstring, that's still be great.

**Review by felixrindt — APPROVED** [review]

File: ephios/core/signup/methods.py (hunk 2 above)
Comment (felixrindt) [reply ephios/core/signup/methods.py:387]: <3

**Pull request merged**